dynamicio/handlers/postgres.py

```diff
@@ -0,0 +1,163 @@
+def session_scope(connection_string: str) -> Generator[SqlAlchemySession, None, None]:
+    """Connect to a database using `connection_string` and returns an active session to that connection.
+
+    Args:
+        connection_string:
+
+    Yields:
+        Active session
+    """
+    engine = create_engine(connection_string)
+    session = Session(bind=engine)
+
+    try:
+        yield session
+        session.commit()
+    except Exception as exc:
+        session.rollback()
+        raise exc
+    finally:
+        session.close()  # pylint: disable=no-member
+
+
+class PostgresCredentials(BaseModel):
+    """..."""
+
+    db_user: str
+    db_password: Optional[str]
+    db_host: str
+    db_port: int
+    db_name: str
+
+    @property
+    def connection_string(self) -> str:
+        """Build connection string out of components."""
+        password = f":{self.db_password}" if self.db_password else ""
+        return f"postgresql://{self.db_user}{password}@{self.db_host}:{self.db_port}/{self.db_name}"
+
+
+class ConfigurationError(Exception):
+    """Raised when configuration is wrong."""
+
+
+class PostgresResource(PostgresCredentials, BaseResource):
+    """...
+
+    Warning: Special case, where even if you explicitly don't validate, but give a pa_schema with strict="filter",
+    the generated sql query will only query specified columns.
+    """
+
+    table_name: Optional[str] = Field(None, description="SQL table name. Needs to be given if no sql_query is given")
+    sql_query: Optional[str] = Field(
+        None, description="SQL query. Will fetch schema defined columns if this is not given."
+    )
+    truncate_and_append: bool = False
+    kwargs: Dict[str, Any] = {}
+
+    def inject(self, **kwargs) -> "PostgresResource":
+        """Inject variables into stuff. Not in place."""
+        clone = deepcopy(self)
+        clone.db_user = inject(clone.db_user, **kwargs)
+        if db_password := self.db_password:
+            clone.db_password = inject(db_password, **kwargs)
+        clone.db_host = inject(clone.db_host, **kwargs)
+        clone.db_name = inject(clone.db_name, **kwargs)
+        if table_name := self.table_name:
+            clone.table_name = inject(table_name, **kwargs)
+        if sql_query := self.sql_query:
+            clone.sql_query = inject(sql_query, **kwargs)
+        return clone
+
+    def _check_injections(self) -> None:
+        """Check that all injections have been completed."""
+        check_injections(self.db_user)
+        if db_password := self.db_password:
+            check_injections(db_password)
+        check_injections(self.db_host)
+        check_injections(self.db_name)
+        if table_name := self.table_name:
+            check_injections(table_name)
+        if sql_query := self.sql_query:
+            check_injections(sql_query)
+
+    def _resource_read(self):
+        """Handles Read operations for Postgres."""
+        if not bool(self.sql_query) ^ bool(self.table_name):
```

Comment:
The `xor` stood out to me. It *is* the more concise way of writing this logic, but I had to think for a bit 🤷

Reply:
I could do this? 
`if not (self.sql_query or self.table_name) and not (self.sql_query and self.table_name):`